pkg/openstack/loadbalancer.go

```diff
@@ -2604,6 +2609,14 @@ func GetLoadBalancerSourceRanges(service *corev1.Service, preferredIPFamily core
 	return ipnets, nil
 }
 
+// isProtocolProxyProtocolSupported checks if the ListenProtocol can be used with the ProxyProtocol
+// for the pool configuration. In the case of UDP listener, Octavia doesn't support the ProxyProtocol
+// and should not be enabled to avoid 403 from Octavia API.
+// https://docs.openstack.org/api-ref/load-balancer/v2/index.html?expanded=create-pool-detail#protocol-combinations-listener-pool
+func isProtocolProxyProtocolSupported(protocol string) bool {
```

Comment:
we already have https://github.com/kubernetes/cloud-provider-openstack/blob/5b36a539124160a070a98e1a23eee9357fc07f2b/pkg/util/util.go#L65-L73 which can be converted into:

```
func Contains[T comparable](list []T, strToSearch T) bool {
        for _, item := range list {
                if item == strToSearch {
                        return true
                }
        }
        return false
}
```